api/v1alpha1/edgeconfig_types.go

```diff
@@ -0,0 +1,185 @@
+	RequiredPrivilegeLevel *RequiredPrivilegeLevel `json:"requiredPrivilegeLevel,omitempty"`
+	// the ansible's playbook options for the playbook
+	Options *AnsibleOptions `json:"ansibleOptions,omitempty"`
+	// the ansible's playbook privilege escalation options for the playbook
+	PrivilegeEscalationOptions *PrivilegeEscalationOptions `json:"privilegeEscalationOptions,omitempty"`
+	//Execution strategy for the playbook
+	ExecutionStrategy ExecutionStrategy `json:"playbooksPriorityMap,omitempty"`
+}
+
+type ExecutionStrategy string
+
+const (
+	StopAtFailuire ExecutionStrategy = "StopOnFailure"
+	RetryOnFailure ExecutionStrategy = "RetryOnFailure"
+	Once           ExecutionStrategy = "ExecuteOnce"
+)
+
+type AnsibleOptions struct {
+	// don't make any changes; instead, try to predict some of the changes that may occur
+	Check bool `json:"check,omitempty"`
+}
+
+type RequiredPrivilegeLevel struct {
+	// See https://man7.org/linux/man-pages/man7/capabilities.7.html
+	CapAdd  []CapType `json:"capAdd,omitempty" description:"Capabilities to add"`
+	CapDrop []CapType `json:"capDrop,omitempty" description:"Capabilities to drop"`
+}
+type PrivilegeEscalationOptions struct {
+	Become bool `json:"become,omitempty"`
+	// +kubebuilder:validation:Enum=sudo;su
+	// +kubebuilder:validation:default=sudo
+	BecomeMethod string `json:"becomeMethod,omitempty"`
+	BecomeUser   string `json:"becomeUser,omitempty"`
+}
+
+type CapType int8
+
+const (
+	CHOWN CapType = iota
+	DAC_OVERRIDE
+	DAC_READ_SEARCH
+	FOWNER
+	FSETID
+	KILL
+	SETGID
+	SETUID
+	SETPCAP
+	LINUX_IMMUTABLE
+	NET_BIND_SERVICE
+	NET_BROADCAST
+	NET_ADMIN
+	NET_RAW
+	IPC_LOCK
+	IPC_OWNER
+	SYS_MODULE
+	SYS_RAWIO
+	SYS_CHROOT
+	SYS_PTRACE
+	SYS_PACCT
+	SYS_ADMIN
+	SYS_BOOT
+	SYS_NICE
+	SYS_RESOURCE
+	SYS_TIME
+	SYS_TTY_CONFIG
+	MKNOD
+	LEASE
+	AUDIT_WRITE
+	AUDIT_CONTROL
+	SETFCAP
+	MAC_OVERRIDE
+	MAC_ADMIN
+	SYSLOG
+	WAKE_ALARM
+	BLOCK_SUSPEND
+	AUDIT_READ
+)
+
+type EdgePlaybookCondition struct {
+	Type EdgePlaybookConditionType `json:"type" description:"type of EdgePlaybookCondition condition"`
+	// Indicates whether that condition is applicable, with possible values "True", "False", or "Unknown"
+	// The absence of a condition should be interpreted the same as Unknown
+	Status metav1.ConditionStatus `json:"status" description:"status of the condition, one of True, False, Unknown"`
+
+	// +optional
+	Reason *string `json:"reason,omitempty" description:"one-word CamelCase reason for the condition's last transition"`
+	// +optional
+	Message *string `json:"message,omitempty" description:"human-readable message indicating details about last transition"`
+	// +optional
+	LastTransitionTime *metav1.Time `json:"lastTransitionTime,omitempty" description:"last time the condition transit from one status to another"`
+}
+
+type EdgePlaybookConditionType string
+
+const (
+	PlaybookDeploying EdgePlaybookConditionType = "Deploying"
+	Completed         EdgePlaybookConditionType = "Completed"
```

Comment:
No, here I'm modelling the fact that an EdgeConfig (with his EdgePlaybook) generate one or more PlaybookExecution.
`Completed` means that the EdgeConfig has been executed on all the involved devices.
If a playbook succeded or failed is modeled in the `PlaybookExecution` because it depends on the specific device  [see example](https://github.com/project-flotta/flotta-operator/blob/3f05bf9ef9c56d1e5b4630b2690a1aedf2934ccc/docs/design/crds.md#status-2).

In future PRs this can be improved (for example, the user can be interested in the percentage of deployments/completions of each EdgeConfig) but I'd like to start with a simple approach so I don't over-complicate this PR.

Makes sense for you?